### DISCUSSION TITLE
Testing computed properties

### DISCUSSION THREAD

**Discussion by arttemiuss43** [livewire/livewire#1145] [1. Help] [answered]
Is there an easy way to test computed properties? I didn't find it in the documentation.

**Comment by asantibanez** [comment]
Hey @arttemiuss43 !

Haven't tried this, but it should work. How about using the computed property in the view and asserting its value using `getViewData('key')`?

**Reply by arttemiuss43** [reply]
Thanks for the suggestion, @asantibanez.

Unfortunately, it didn't work. It gives me the following error: `ErrorException: Undefined index: variation
`. 
Here is my setup:
```php
// App\Http\Livewire\Product

public function getVariationProperty()
{
    return app(VariationsRepository::class)
        ->findBySku($this->sku, array_values($this->activeProperties));
}
```

```php
// tests/Feature/VariationTest

/** @test */
public function it_returns_variation_by_sku()
{
    $variation = tap(Variation::first())->update([
        'sku' => 123456
    ]);

    $component = Livewire::test(Product::class, ['sku' => '123456']);

    $this->assertEquals(123456, $component->viewData('variation')->sku);
}
```

**Comment by jhoff** [comment]
Try `$component->viewData('_instance')->variation` to get the value of a computed property.

Full disclaimer: I don't know if this is a proper or full solution, but it worked for my simple test.

**Comment by RichardStyles** [comment]
For anyone else needing this, there is a slightly cleaner way.

```
$component->get('variation');
```

Source: 
https://github.com/livewire/livewire/blob/fa512db9bf4ee0b0a23812a5073854a84d15eff8/src/Features/SupportTesting/Testable.php#L86-L89

So this will get the value from a computed property.

**Reply by joshhanley** [reply, marked as answer]
@RichardStyles you can just call the computed property name to test it:

Directly in an assert
```php
Livewire::test(MyComponent::class)
    ->assertSet('myComputedProperty', 1234);
```

Or access it's value
```php
$component = Livewire::test(MyComponent::class);

$this->assertEquals(1234, $component->myComputedProperty);
```
